src/SdkCommon/ClientRuntime/ClientRuntime/ExtensibleEnum.cs

```diff
@@ -5,73 +5,77 @@
     /// </summary>
-    public abstract class ExtensibleEnum<T> : IEquatable<T> where T : ExtensibleEnum<T>
+    public abstract class ExtensibleEnum<T, V> : IEquatable<T> where T : ExtensibleEnum<T, V>
     {
-        private static readonly Lazy<Dictionary<string, T>> _nameMap =
-            new Lazy<Dictionary<string, T>>(CreateNameMap, isThreadSafe: true);
+        private static readonly ConcurrentDictionary<V, T> _valueMap =
+            new ConcurrentDictionary<V, T>(CreateValueMap());
 
-        private string _name;
+        private V _value;
 
         /// <summary>
         /// Initializes a new instance of the ExtensibleEnum class.
         /// </summary>
-        /// <param name="name">The value for this instance of the extensible enumeration.</param>
-        protected ExtensibleEnum(string name)
+        /// <param value="value">The value for this instance of the extensible enumeration.</param>
+        protected ExtensibleEnum(V value)
         {
-            _name = name;
+            _value = value;
         }
 
         /// <summary>
         /// Compares two ExtensibleEnum values for equality.
         /// </summary>
-        /// <param name="lhs">The first ExtensibleEnum to compare.</param>
-        /// <param name="rhs">The second ExtensibleEnum to compare.</param>
+        /// <param value="lhs">The first ExtensibleEnum to compare.</param>
+        /// <param value="rhs">The second ExtensibleEnum to compare.</param>
         /// <returns>true if the ExtensibleEnum objects are equal or are both null; false otherwise.</returns>
-        public static bool operator ==(ExtensibleEnum<T> lhs, T rhs) => object.Equals(lhs, rhs);
+        public static bool operator ==(ExtensibleEnum<T, V> lhs, T rhs) => object.Equals(lhs, rhs);
 
         /// <summary>
         /// Compares two ExtensibleEnum values for inequality.
         /// </summary>
-        /// <param name="lhs">The first ExtensibleEnum to compare.</param>
-        /// <param name="rhs">The second ExtensibleEnum to compare.</param>
+        /// <param value="lhs">The first ExtensibleEnum to compare.</param>
+        /// <param value="rhs">The second ExtensibleEnum to compare.</param>
         /// <returns>true if the ExtensibleEnum objects are not equal; false otherwise.</returns>
-        public static bool operator !=(ExtensibleEnum<T> lhs, T rhs) => !object.Equals(lhs, rhs);
+        public static bool operator !=(ExtensibleEnum<T, V> lhs, T rhs) => !object.Equals(lhs, rhs);
 
         /// <summary>
         /// Defines explicit conversion from ExtensibleEnum to string.
         /// </summary>
-        /// <param name="name">ExtensibleEnum to convert.</param>
+        /// <param val="val">ExtensibleEnum to convert.</param>
         /// <returns>The ExtensibleEnum as a string.</returns>
-        public static explicit operator string(ExtensibleEnum<T> name) => name?.ToString();
+        public static explicit operator string(ExtensibleEnum<T, V> val) => val?.ToString();
```

Comment:
The purpose of this conversion operator was to "unwrap" extensible enum values to their underlying types, so I guess this should be `operator V` now.